gtsfm/frontend/correspondence_generator/keypoint_aggregator/keypoint_aggregator_dedup.py

```diff
@@ -46,8 +51,8 @@ def append_unique_keypoints(
         for k, uv in enumerate(keypoints.coordinates):
             diff_norms = np.linalg.norm(per_image_kpt_coordinates[i] - uv, axis=1)
             # TODO(johnwlambert,ayushbaid): test loosening threshold below to some epsilon.
-            is_identical = np.any(diff_norms == 0)
-            if len(per_image_kpt_coordinates[i]) > 0 and is_identical:
+            is_duplicate = np.any(diff_norms <= self.nms_radius)
+            if len(per_image_kpt_coordinates[i]) > 0 and is_duplicate:
```

Comment:
Maybe we should use the average coordinate?

Reply:
Left a TODO, thanks.